### PR TITLE
Add integration tests to workflows

### PR DESCRIPTION
**Pull request by DakCrowder**
Makes the integration tests run via adding a gh workflow.

See the run for this PR [here](https://github.com/flightctl/flightctl-ansible/actions/runs/12146044622/job/33869040913?pr=14)

The way the tests are run is by checking out the flightctl repository, setting up dependencies, and then running a `make deploy` to get the services spun up.  This is a bit heavy handed (we are installing more dependencies than we need) and could be streamlined in the future but is the easiest way to get up and running.

Notes:
- A make utility helper has been added to update the config file the integration tests use to point to the flightctl api service by looking up the service address from the flightctl yaml config.  Ansbile integration tests don't have a great way to pass along config to tests dynamically so writing to this file was the easiest way for now
- Modified the integration tests to run locally in containers which is recommended and how the sanity/unit tests were already configured

### CODE REVIEW

File: .github/workflows/integration-tests.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  cancel-in-progress: true
+
+jobs:
+  integration-tests:
+    runs-on: "ubuntu-latest"
+    strategy:
+      matrix:
+        versions:
+          - ansible: stable-2.16
+            python: "3.10"
```
Comment (amir-yogev-gh) [inline .github/workflows/integration-tests.yaml:20]: Can we use a more recent Python version (e.g. 3.13)?
Are there any limitations from Ansible regarding this?

Comment (DakCrowder) [reply .github/workflows/integration-tests.yaml:20]: Based on the ansible core version we are (currently) slated to support (2.16) it is compatible with python 3.10-3.12, I believe ansible core supports the 3 latest python versions at time of release.  Eventually as part of the release preparation for Galaxy/Automation Hub we'll want to modify the existing CI tests to run against a matrix of these different python versions.
https://docs.ansible.com/ansible/latest/reference_appendices/release_and_maintenance.html#ansible-core-control-node-python-support

Looking at that link though I wonder if we should look to upgrade our "default" ansible core version to 2.17 or 2.18 as it buys us much more time before EoL and support for newer python versions.

File: README.md [hunk 2]
```diff
@@ -142,10 +145,12 @@ Run locally via `make test-sanity`
 
 Integration tests are dependent on:
 - A flightctl instance the tests can hit
-  - Locally the easiest way is to run `make deploy` from the main flightctl repository
 - The flightctl_host var inside integration_config.yml set to the running flightctl api service
 
-Run locally via `make test-integration`
+The easiest way to run tests locally is to:
+- Run `make deploy` from the main flightctl repository
+- Run `make write-integration-config` from this repository to create the proper integration config from your running services
```
Comment (hexfusion) [inline README.md:152]: question: is this something a user would use to manipulate an integration test? I guess the main question is can we just bundle this into `make test-integration`

Comment (hexfusion) [reply README.md:152]: also in flightctl we use <type>-test I also see `ansible-test`

File: Makefile [hunk 3]
```diff
@@ -5,7 +5,11 @@ test-unit:
 	ansible-test units --docker -v --color --python $(PYTHON_VERSION) $(?TEST_ARGS)
 
 test-integration:
-	ansible-test integration --diff --no-temp-workdir --color --python $(PYTHON_VERSION) -v $(?TEST_ARGS)
+	ansible-test integration --docker --diff --color --python $(PYTHON_VERSION) -v $(?TEST_ARGS)
```
Comment (hexfusion) [inline Makefile:8]: not podman :))) ?

File: README.md (hunk 2 above)
Comment (DakCrowder) [reply README.md:152]: Yeah thats a good point, i should just bundle them both.  And can rename to be more inline w/ the -test postfix

File: Makefile (hunk 3 above)
Comment (DakCrowder) [reply Makefile:8]: --docker option falls back to podman - https://docs.ansible.com/ansible/latest/dev_guide/testing_running_locally.html#containers.  Maybe we can get the --docker option namechanged at some point :D

**Pull request merged**